lib/pbench/server/api/resources/endpoint_configure.py

```diff
@@ -158,10 +158,19 @@ def get(self):
                     },
                 }
                 url = self.param_template.sub("", url)
-                path = url[len(self.uri_prefix) + 1 :]
-                path = path.replace("/", "_")
-                apis[path] = urljoin(host, url)
-                templates[path] = template
+                path = rule.endpoint
+
+                # We have some URI endpoints that repeat a basic URI pattern.
+                # The "primary" may have several URI parameters; the others
+                # have fewer parameters (e.g., "/x/{p}" and "/x" or
+                # "/x/{p}/{n}" and "/x/{p}" and "/x") and won't capture all
+                # the information we want. So we only keep the variant with the
+                # highest parameter count.
+                if path not in templates or (
+                    len(template["params"]) > len(templates[path]["params"])
+                ):
+                    apis[path] = urljoin(host, url)
+                    templates[path] = template
```

Comment:
You prefer the conditional/searching/dynamic approach over just requiring that the "right" one be listed first (or having a "primary" and optional "aliases", or some other way to distinguish it)?  [You've mentioned elsewhere that you're concerned about Flask messing with the ordering, which makes good sense; however, having some explicit pointer to the primary might be better than trying to solve it heuristically.]

Reply:
Nope; this is good.